lib/block_view_dao_coin_limit_order.go

```diff
@@ -0,0 +1,1763 @@
+func (bav *UtxoView) _getNextLimitOrdersToFill(
+	transactorOrder *DAOCoinLimitOrderEntry, lastSeenOrder *DAOCoinLimitOrderEntry) (
+	[]*DAOCoinLimitOrderEntry, error) {
+	// Get matching limit order entries from database.
+	matchingOrders, err := bav.GetDbAdapter().GetMatchingDAOCoinLimitOrders(transactorOrder, lastSeenOrder)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update UTXO with relevant limit order entries from database.
+	for _, matchingOrder := range matchingOrders {
+		if _, exists := bav.DAOCoinLimitOrderMapKeyToDAOCoinLimitOrderEntry[matchingOrder.ToMapKey()]; !exists {
+			bav._setDAOCoinLimitOrderEntryMappings(matchingOrder)
+		}
+	}
+
+	// Aggregate all matching orders then sort.
+	sortedMatchingOrders := []*DAOCoinLimitOrderEntry{}
+
+	// Aggregate matching orders.
+	for _, matchingOrder := range bav.DAOCoinLimitOrderMapKeyToDAOCoinLimitOrderEntry {
+		// This doesn't mean that the matching order is invalid and should be deleted.
+		// It just means that the matching order isn't actually a viable match.
+		err := bav.IsValidDAOCoinLimitOrderMatch(transactorOrder, matchingOrder)
+		if err != nil {
+			// If matching own order, fail immediately. Otherwise just skip this order.
+			if err == RuleErrorDAOCoinLimitOrderMatchingOwnOrder {
+				return nil, err
+			}
+
+			continue
+		}
+
+		// We should have seen this order already.
+		if lastSeenOrder != nil && !lastSeenOrder.IsBetterMatchingOrderThan(matchingOrder) {
+			continue
+		}
+
+		sortedMatchingOrders = append(sortedMatchingOrders, matchingOrder)
+	}
+
+	// Sort matching orders by best matching.
+	// Sort logic first looks at price, then block height (FIFO), then quantity (lowest first).
+	sort.Slice(sortedMatchingOrders, func(ii, jj int) bool {
+		return sortedMatchingOrders[ii].IsBetterMatchingOrderThan(sortedMatchingOrders[jj])
+	})
+
+	// Pull orders up to when the quantity is filled or we run out of orders.
+	outputMatchingOrders := []*DAOCoinLimitOrderEntry{}
+	transactorOrderBuyingQuantity := transactorOrder.QuantityToFillInBaseUnits
+
+	for _, matchingOrder := range sortedMatchingOrders {
+		outputMatchingOrders = append(outputMatchingOrders, matchingOrder)
+
+		// To properly compare quantities, we need to compare the quantity
+		// that the transactor order is interested in buying to the quantity
+		// the matching order is interested in selling.
+		matchingOrderSellingQuantity, err := matchingOrder.BaseUnitsToSellUint256()
+		if err != nil {
+			// This should never happen as we validate the
+			// stored orders when they are submitted.
+			return nil, err
+		}
+
+		// Break once the transactor's buying quantity is <= this matching
+		// order's selling quantity or their buying quantity is <= 0.
+		if transactorOrderBuyingQuantity.Eq(matchingOrderSellingQuantity) ||
+			transactorOrderBuyingQuantity.Lt(matchingOrderSellingQuantity) ||
+			transactorOrderBuyingQuantity.LtUint64(1) {
+			break
+		}
```

Comment:
I don't think this is quite right. This needs to be updated to use the `_calculateDAOCoinsTransferredInLimitOrderMatch` function. The `transactorOrder.QuantityToFillInBaseUnits` can be a selling quantity if they submitted an ASK order and is not guaranteed to be a buying quantity. I can make that change.